[2024-12-23 22:33:43,375][flwr][INFO] - Starting Flower simulation, config: num_rounds=10, no round_timeout
[2024-12-23 22:33:43,429][filelock][INFO] - Lock 125085530339168 acquired on /tmp/ray/session_2024-12-23_22-33-43_429309_1151706/node_ip_address.json.lock
[2024-12-23 22:33:43,430][filelock][INFO] - Lock 125085530339168 released on /tmp/ray/session_2024-12-23_22-33-43_429309_1151706/node_ip_address.json.lock
[2024-12-23 22:33:43,430][filelock][INFO] - Lock 125085530339168 acquired on /tmp/ray/session_2024-12-23_22-33-43_429309_1151706/ports_by_node.json.lock
[2024-12-23 22:33:43,430][filelock][INFO] - Lock 125085530339168 released on /tmp/ray/session_2024-12-23_22-33-43_429309_1151706/ports_by_node.json.lock
[2024-12-23 22:33:43,430][filelock][INFO] - Lock 125085530339168 acquired on /tmp/ray/session_2024-12-23_22-33-43_429309_1151706/ports_by_node.json.lock
[2024-12-23 22:33:43,430][filelock][INFO] - Lock 125085530339168 released on /tmp/ray/session_2024-12-23_22-33-43_429309_1151706/ports_by_node.json.lock
[2024-12-23 22:33:43,430][filelock][INFO] - Lock 125085530339168 acquired on /tmp/ray/session_2024-12-23_22-33-43_429309_1151706/ports_by_node.json.lock
[2024-12-23 22:33:43,431][filelock][INFO] - Lock 125085530339168 released on /tmp/ray/session_2024-12-23_22-33-43_429309_1151706/ports_by_node.json.lock
[2024-12-23 22:33:43,431][filelock][INFO] - Lock 125085529989024 acquired on /tmp/ray/session_2024-12-23_22-33-43_429309_1151706/ports_by_node.json.lock
[2024-12-23 22:33:43,431][filelock][INFO] - Lock 125085529989024 released on /tmp/ray/session_2024-12-23_22-33-43_429309_1151706/ports_by_node.json.lock
[2024-12-23 22:33:43,431][filelock][INFO] - Lock 125085529988064 acquired on /tmp/ray/session_2024-12-23_22-33-43_429309_1151706/ports_by_node.json.lock
[2024-12-23 22:33:43,431][filelock][INFO] - Lock 125085529988064 released on /tmp/ray/session_2024-12-23_22-33-43_429309_1151706/ports_by_node.json.lock
[2024-12-23 22:33:45,574][flwr][INFO] - Flower VCE: Ray initialized with resources: {'node:__internal_head__': 1.0, 'object_store_memory': 2352775987.0, 'memory': 4705551975.0, 'CPU': 12.0, 'node:192.168.1.11': 1.0}
[2024-12-23 22:33:45,574][flwr][INFO] - Optimize your simulation with Flower VCE: https://flower.ai/docs/framework/how-to-run-simulations.html
[2024-12-23 22:33:45,574][flwr][INFO] - Flower VCE: Resources for each Virtual Client: {'num_cpus': 2, 'num_gpus': 0}
[2024-12-23 22:33:45,584][flwr][INFO] - Flower VCE: Creating VirtualClientEngineActorPool with 6 actors
[2024-12-23 22:33:45,585][flwr][INFO] - [INIT]
[2024-12-23 22:33:45,586][flwr][INFO] - Requesting initial parameters from one random client
[2024-12-23 22:33:47,659][flwr][INFO] - Received initial parameters from one random client
[2024-12-23 22:33:47,660][flwr][INFO] - Starting evaluation of initial global parameters
[2024-12-23 22:33:47,687][flwr][ERROR] - index 0 is out of bounds for dimension 0 with size 0
[2024-12-23 22:33:47,688][flwr][ERROR] - Traceback (most recent call last):
  File "/home/vadi/anaconda3/lib/python3.8/site-packages/flwr/simulation/app.py", line 339, in start_simulation
    hist = run_fl(
  File "/home/vadi/anaconda3/lib/python3.8/site-packages/flwr/server/server.py", line 492, in run_fl
    hist, elapsed_time = server.fit(
  File "/home/vadi/anaconda3/lib/python3.8/site-packages/flwr/server/server.py", line 95, in fit
    res = self.strategy.evaluate(0, parameters=self.parameters)
  File "/home/vadi/anaconda3/lib/python3.8/site-packages/flwr/server/strategy/fedavg.py", line 167, in evaluate
    eval_res = self.evaluate_fn(server_round, parameters_ndarrays, {})
  File "/home/vadi/oii/Academics/BtechProject/tutorial/server.py", line 30, in evaluate_fn
    model.load_state_dict(state_dict, strict=True)
  File "/home/vadi/anaconda3/lib/python3.8/site-packages/torch/nn/modules/module.py", line 2201, in load_state_dict
    load(self, state_dict)
  File "/home/vadi/anaconda3/lib/python3.8/site-packages/torch/nn/modules/module.py", line 2189, in load
    load(child, child_state_dict, child_prefix)  # noqa: F821
  File "/home/vadi/anaconda3/lib/python3.8/site-packages/torch/nn/modules/module.py", line 2183, in load
    module._load_from_state_dict(
  File "/home/vadi/anaconda3/lib/python3.8/site-packages/torch/nn/modules/batchnorm.py", line 115, in _load_from_state_dict
    super()._load_from_state_dict(
  File "/home/vadi/anaconda3/lib/python3.8/site-packages/torch/nn/modules/module.py", line 2059, in _load_from_state_dict
    input_param = input_param[0]
IndexError: index 0 is out of bounds for dimension 0 with size 0

[2024-12-23 22:33:47,688][flwr][ERROR] - Your simulation crashed :(. This could be because of several reasons. The most common are: 
	 > Sometimes, issues in the simulation code itself can cause crashes. It's always a good idea to double-check your code for any potential bugs or inconsistencies that might be contributing to the problem. For example: 
		 - You might be using a class attribute in your clients that hasn't been defined.
		 - There could be an incorrect method call to a 3rd party library (e.g., PyTorch).
		 - The return types of methods in your clients/strategies might be incorrect.
	 > Your system couldn't fit a single VirtualClient: try lowering `client_resources`.
	 > All the actors in your pool crashed. This could be because: 
		 - You clients hit an out-of-memory (OOM) error and actors couldn't recover from it. Try launching your simulation with more generous `client_resources` setting (i.e. it seems {'num_cpus': 2, 'num_gpus': 0} is not enough for your run). Use fewer concurrent actors. 
		 - You were running a multi-node simulation and all worker nodes disconnected. The head node might still be alive but cannot accommodate any actor with resources: {'num_cpus': 2, 'num_gpus': 0}.
Take a look at the Flower simulation examples for guidance <https://flower.ai/docs/framework/how-to-run-simulations.html>.
